Orange/evaluation/testing.py

```diff
@@ -462,65 +415,8 @@ def __init__(self, data, learners, n_resamples=10, train_size=None,
                 test_size=self.test_size, random_state=self.random_state
             )
 
-        self.folds = []
-        if self.store_models:
-            self.models = []
-
-        row_indices = []
-        actual = []
-        predicted = [[] for _ in learners]
-        probabilities = [[] for _ in learners]
-        fold_start = 0
-        nmethods = len(learners)
-        n_callbacks = self.n_resamples * nmethods
-        for samp_idx, (train, test) in enumerate(self.indices):
-            train_data, test_data = data[train], data[test]
-            if preprocessor is not None:
-                train_data = self.preprocessor(train_data)
-            self.folds.append(slice(fold_start, fold_start + len(test)))
-            row_indices.append(test)
-            actual.append(test_data.Y.flatten())
-            if self.store_models:
-                fold_models = [None] * nmethods
-                self.models.append(fold_models)
-
-            for i, learner in enumerate(learners):
-                model = self.train_if_succ(i, learner, train_data)
-                self.call_callback((samp_idx * nmethods + i ) / n_callbacks)
-                if model:
-                    if self.store_models:
-                        fold_models[i] = model
-                    if data.domain.has_discrete_class:
-                        values, probs = model(test_data, model.ValueProbs)
-                        predicted[i].append(values)
-                        probabilities[i].append(probs)
-                    elif data.domain.has_continuous_class:
-                        values = model(test_data, model.Value)
-                        predicted[i].append(values)
-                else:
-                    predicted[i].append(np.zeros((len(test_data),)))
-                    if data.domain.has_discrete_class:
-                        probabilities[i].append(
-                            np.zeros((len(test_data),
-                                      len(data.domain.class_var.values))))
-
-            fold_start += len(test)
-
-        row_indices = np.hstack(row_indices)
-        actual = np.hstack(actual)
-        predicted = np.array([np.hstack(pred) for pred in predicted])
-        if data.domain.has_discrete_class:
-            probabilities = np.array([np.vstack(prob) for prob in probabilities])
-        nrows = len(actual)
-        nmodels = len(predicted)
-
-        self.nrows = len(actual)
-        self.row_indices = row_indices
-        self.actual = actual
-        self.predicted = predicted.reshape(nmodels, nrows)
-        if data.domain.has_discrete_class:
-            self.probabilities = probabilities
-        self.call_callback(1)
+        self.prepare_arrays(data)
+        self.fit(data)
 
 
 class TestOnTestData(Results):
```

Comment:
Could this class also make use of `Results.prepare_arrays()` and `Results.fit()`? Perhaps by storing `self._train_data` and `self._test_data` (both equal to `data` in e.g. `CrossValidation`) and setting:
`self.indices = ((Ellipses, Ellipses),)`?